modyn/tests/cpp_test_main.cpp

```diff
@@ -1,6 +1,23 @@
-#include "gtest/gtest.h"
+#include <gtest/gtest.h>
+
+#include <filesystem>
+#include <iostream>
+
+#include "test_utils.hpp"
+
+// Global fixtures run before unit tests are executed
+class GlobalTestEnvironment : public ::testing::Environment {
```

Comment:
Could we not move this logic into `get_tmp_testdir`?  Then we do not need the global test environment. Although I am not sure what the cleaner option is here. I am just wondering because currently, all tests have their own subdirectory (`/tmp/user/bla`). The tests need to create that directory anyways, and hence could also just recursively create `/tmp/user` if it does not exist. In this case, we would not to create the /tmp/user directory explicitly. Maybe I am also missing something^^

Reply:
I don't have a strong preference for where to put this logic. Making a couple of extra `create_directory` isn't harmful within the tests. Coming from a Python background I would have searched for some global fixture in a conftest file. I guess `::testing::Environment` is the closest thing to that in C++. 
We can't rely on recursive parent construction though:

> the parent directory must already exist

https://en.cppreference.com/w/cpp/filesystem/create_directory

Reply:
Okay, if we need to create the parent_directory explicitly anyways, this is okay :-) Thank you!